cmd/controller/main.go

```diff
@@ -43,6 +52,22 @@ func main() {
 	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
 	cloudProvider := metrics.Decorate(awsCloudProvider)
 
+	cm := &v1.ConfigMap{}
+	if err := op.GetClient().Get(ctx, types.NamespacedName{Name: "karpenter-global-settings"}, cm); err != nil {
+		logging.FromContext(ctx).Error(err)
+	}
+
+	serverVersion, err := op.KubernetesInterface.Discovery().ServerVersion()
```

Comment:
We already have support for doing something like this in the AMI provider. What if we just created a version provider that provided us this info for free?